src/dbnode/persist/fs/seek.go

```diff
@@ -195,26 +177,34 @@ func (s *seeker) Open(namespace ident.ID, shard uint32, blockStart time.Time) er
 	}
 
 	// Setup digest readers
-	s.infoFdWithDigest.Reset(infoFd)
-	s.indexFdWithDigest.Reset(indexFd)
-	s.summariesFdWithDigest.Reset(summariesFd)
-	s.digestFdWithDigestContents.Reset(digestFd)
-
+	var (
+		infoFdWithDigest           = digest.NewFdWithDigestReader(s.opts.infoBufferSize)
+		indexFdWithDigest          = digest.NewFdWithDigestReader(s.opts.dataBufferSize)
+		bloomFilterFdWithDigest    = digest.NewFdWithDigestReader(s.opts.dataBufferSize)
+		summariesFdWithDigest      = digest.NewFdWithDigestReader(s.opts.dataBufferSize)
+		digestFdWithDigestContents = digest.NewFdWithDigestContentsReader(s.opts.infoBufferSize)
+	)
 	defer func() {
-		// NB(r): We don't need to keep these FDs open as we use these up front
-		s.infoFdWithDigest.Close()
-		s.indexFdWithDigest.Close()
-		s.bloomFilterFdWithDigest.Close()
-		s.digestFdWithDigestContents.Close()
+		// NB(rartoul): We don't need to keep these FDs open as we use these up front
+		infoFdWithDigest.Close()
```

Comment:
Looks like this closes the underlying fd too. Is the `mmap` valid after the fd is closed?

Reply:
Yeah its safe to close an FD once the mmap syscall has executed, it wont affect the mmap (this is what we were already doing also)

Reply:
👍 sounds good